phoenix-core/src/it/java/org/apache/phoenix/end2end/index/MutableIndexReplicationIT.java

```diff
@@ -274,8 +274,133 @@ private boolean ensureAnyRows(HTable remoteTable) throws IOException {
         return found;
     }
 
+    private boolean ensureNumberOfRows(HTable remoteTable, int numRows) throws IOException {
+        Scan scan = new Scan();
+        scan.setRaw(true);
+        ResultScanner scanner = remoteTable.getScanner(scan);
+        int rowsFound = 0;
+        for (Result r : scanner) {
+            LOG.info("got row: " + r);
+            rowsFound++;
+        }
+        scanner.close();
+        return (numRows == rowsFound);
+    }
+
     private static Connection getConnection() throws Exception {
         Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
         return DriverManager.getConnection(URL, props);
     }
+
+    @Test
+    public void testReplicationWithMutableLocalIndexes() throws Exception {
+        // Generating new table names so they don't conflict with previous or concurrent test runs.
+        String dataTableName = generateUniqueName();
+        String dataTableFullName = SchemaUtil.getTableName(SCHEMA_NAME, dataTableName);
+        String indexTableName = generateUniqueIndexName();
```

Comment:
@hnguyen08 Is there a reason you can't just use generateUniqueName() for the indexTableName?